src/plugins/summaryWidget/src/ConditionEvaluator.js

```diff
@@ -0,0 +1,449 @@
+                },
+                text: 'text ends with',
+                appliesTo: ['string'],
+                inputCount: 1,
+                getDescription: function (values) {
+                    return ' ends with ' + values[0];
+                }
+            },
+            textIsExactly: {
+                operation: function (input) {
+                    return input[0] === input[1];
+                },
+                text: 'text is exactly',
+                appliesTo: ['string'],
+                inputCount: 1,
+                getDescription: function (values) {
+                    return ' is exactly ' + values[0];
+                }
+            },
+            isUndefined: {
+                operation: function (input) {
+                    return typeof input[0] === 'undefined';
+                },
+                text: 'is undefined',
+                appliesTo: ['string', 'number'],
+                inputCount: 0,
+                getDescription: function () {
+                    return ' is undefined';
+                }
+            }
+        };
+    }
+
+    /**
+     * Evaluate the conditions passed in as an argument, and return the boolean
+     * value of these conditions. Available evaluation modes are 'any', which will
+     * return true if any of the conditions evaluates to true (i.e. logical OR); 'all',
+     * which returns true only if all conditions evalute to true (i.e. logical AND);
+     * or 'js', which returns the boolean value of a custom JavaScript conditional.
+     * @param {} conditions Either an array of objects with object, key, operation,
+     *                      and value fields, or a string representing a JavaScript
+     *                      condition.
+     * @param {string} mode The key of the mode to use when evaluating the conditions.
+     * @return {boolean} The boolean value of the conditions
+     */
+    ConditionEvaluator.prototype.execute = function (conditions, mode) {
+        var active = false,
+            conditionValue,
+            conditionDefined = false,
+            self = this,
+            firstRuleEvaluated = false,
+            compositionObjs = this.compositionObjs;
+
+        if (mode === 'js') {
+            active = this.executeJavaScriptCondition(conditions);
+        } else {
+            (conditions || []).forEach(function (condition) {
+                conditionDefined = false;
+                if (condition.object === 'any') {
+                    conditionValue = false;
+                    Object.keys(compositionObjs).forEach(function (objId) {
+                        try {
+                            conditionValue = conditionValue ||
+                                self.executeCondition(objId, condition.key,
+                                    condition.operation, condition.values);
+                            conditionDefined = true;
+                        } catch (e) {
+                            //ignore a malformed condition
```

Comment:
This might be a confusing word choice on my part. The evaluator considers conditions 'malformed' if they are only partially configured, so it regularly encounters them in normal usage. So, for example, if the user has chosen a source object and nothing else in a condition, the condition is still passed to the evaluator, but ignored.